FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_DNS.c

```diff
@@ -828,29 +1201,30 @@ TickType_t uxWriteTimeOut_ticks = ipconfigDNS_SEND_BLOCK_TIME_TICKS;
 /*-----------------------------------------------------------*/
 
 _static size_t prvCreateDNSMessage( uint8_t *pucUDPPayloadBuffer,
-								   const char *pcHostName,
-								   TickType_t uxIdentifier )
+									const char *pcHostName,
+									TickType_t uxIdentifier,
+									UBaseType_t uxHostType )
 {
 DNSMessage_t *pxDNSMessageHeader;
 uint8_t *pucStart, *pucByte;
 DNSTail_t const * pxTail;
 static const DNSMessage_t xDefaultPartDNSHeader =
 {
-	0,                 /* The identifier will be overwritten. */
-	dnsOUTGOING_FLAGS, /* Flags set for standard query. */
-	dnsONE_QUESTION,   /* One question is being asked. */
-	0,                 /* No replies are included. */
-	0,                 /* No authorities. */
-	0                  /* No additional authorities. */
+	0,					/* The identifier will be overwritten. */
+	dnsOUTGOING_FLAGS,	/* Flags set for standard query. */
+	dnsONE_QUESTION,	/* One question is being asked. */
+	0,					/* No replies are included. */
+	0,					/* No authorities. */
+	0					/* No additional authorities. */
 };
 
 	/* Copy in the const part of the header. Intentionally using different
 	 * pointers with memcpy() to put the information in to correct place. */
-	( void ) memcpy( ( void * ) pucUDPPayloadBuffer, ( const void * ) ( &( xDefaultPartDNSHeader ) ), sizeof( xDefaultPartDNSHeader ) );
+	( void ) memcpy( pucUDPPayloadBuffer, &( xDefaultPartDNSHeader ), sizeof( xDefaultPartDNSHeader ) );
 
 	/* Write in a unique identifier. Cast the Payload Buffer to DNSMessage_t
 	 * to easily access fields of the DNS Message. */
-	pxDNSMessageHeader = ipCAST_PTR_TO_TYPE_PTR( DNSMessage_t, pucUDPPayloadBuffer );
+	pxDNSMessageHeader = ipPOINTER_CAST( DNSMessage_t *, pucUDPPayloadBuffer );
```

Comment:
Can we keep using `ipCAST_PTR_TO_TYPE_PTR`? And do the same in all the instances? See my "Instance" comments.

Reply:
Done as well.

Reply:
>  And do the same in all the instances? See my "Instance" comments.

I grep the work "instance", but I don't see it. What do you mean exactly?